org.eclipse.lsp4xml.extensions.maven/src/main/java/org/eclipse/lsp4xml/extensions/maven/searcher/RemoteRepositoryIndexSearcher.java

```diff
@@ -281,4 +291,65 @@ private void initializeContext()
 
 	}
 
+	public void closeContext() {
+		for (IndexingContext context : indexingContexts.values()) {
+			try {
+				indexer.closeIndexingContext(context, false);
+			} catch (IOException e) {
+				System.out.println("Warning - could not close context: " + context.getId());
+				e.printStackTrace();
+			}
+		}
+
+	}
+
+//////////////////////////////////// Helper Methods ///////////////////////////////////////////////
+
+	private CompletableFuture<List<ArtifactInfo>> createIndexerQuery(Artifact artifactToSearch,
```

Comment:
This function was factored from getArtifactVersions(), the logic itself is not new